### DISCUSSION TITLE
Migrate a cluster datastore from embedded sqlite to etcd/external mysql

### DISCUSSION THREAD

**Discussion by longnsp** [k3s-io/k3s#7936] [Q&A] [answered]
- I have a cluster with 3 nodes, 1 master and 2 workers. As we are still new to Kubernetes as a whole, I installed k3s with embedded sqlite as datastore (and now I realized should had at least use etcd). Now as that cluster having more and more deployments I'm kinda worried about backup and restore the cluster's database. The backup for sqlite is actually quite simple as the doc said you just need to take a copy of `/var/lib/rancher/k3s/server/db/`.
- I have read the doc [here](https://docs.k3s.io/datastore/ha-embedded) that said if I have an existing cluster using the default embedded SQLite database, I can convert it to etcd by simply restarting the k3s server with the `--cluster-init` flag. How do I do it exactly here and is it safe for my cluster?
- So my question is can I change the datastore of my running cluster from sqlite to another and if yes can you provide some general instructions. Here is the nodes k3s version, if you need any more infos just tell me.
![image](https://github.com/k3s-io/k3s/assets/99666567/a12f08fc-3919-4986-a61f-8c5e1fddc697)

**Comment by brandond** [comment, marked as answer]
The general instructions are as you quoted - add `--cluster-init` to the server's CLI flags (control-plane) node, and restart it. The best place to do this is usually in the k3s systemd unit - edit the service file, use `systemctl daemon-reload` to reload it, then restart the service.

This will convert your datastore from sqlite to etcd. This is the only supported datastore conversion.
